qiskit/test/mock/fake_backend.py

```diff
@@ -38,6 +62,257 @@ def __init__(self, token="123456", url="https://"):
         self.project = "project"
 
 
+class FakeBackendV2(BackendV2):
+    """This is a dummy bakend just for resting purposes. the FakeBackendV2 builds on top of the BackendV2 base class."""
+
+    dirname = None
+    conf_filename = None
+    props_filename = None
+    defs_filename = None
+    backend_name = None
+
+    def __init__(self):
+        self._conf_dict = self._get_conf_dict_from_json()
+        self._props_dict = self._set_props_dict_from_json()
+        self._defs_dict = self._set_defs_dict_from_json()
+        super().__init__(
+            provider=None,
+            name=self._conf_dict.get("backend_name"),
+            description=self._conf_dict.get("description"),
+            online_date=self._conf_dict.get("online_date"),
+            backend_version=self._conf_dict.get("backend_version"),
+        )
+        self._target = None
+        self._qubit_properties = None
+
+    def _get_conf_dict_from_json(self) -> dict:
+        if not self.conf_filename:
+            warnings.warn("No configuration file has been defined", UserWarning)
+            return None
+        conf_dict = self._load_json(self.conf_filename)
+        decode_backend_configuration(conf_dict)
+        conf_dict["backend_name"] = self.backend_name
+        return conf_dict
+
+    def _set_props_dict_from_json(self) -> dict:
+        if not self.props_filename:
+            warnings.warn("No properties file has been defined", UserWarning)
+            return None
```

Comment:
I don't think we need to warn here (or for any of the other payloads) as this should be fixed in the content of the project. If a warning is raised it will always be raised for that fake backend and there is nothing a user can do about it. We should just return None here and not warn

Reply:
I added this method based on the `fake_qasm_backend` where a `QiskitError` is raised when properties file do not exist. https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/test/mock/fake_qasm_backend.py#L58-L60

I change it from error to warning exactly because I agree with you that nothing a user can do about that :) I can remove the warning. Is there a way users can know whether a certain backend has conf, props and defs files without looking at the source code? I believe users may want to know that because lacking of a certain file may impact the accuracy of simulation. Am I right?

Reply:
Missing properties means it will be simulated without noise, missing defaults means we can't run pulse simulations (and won't have the default calibrations set for instructions). Missing all configuration and defaults means the backend won't work :) 

If a backend is missing properties I'd question adding it to the set of BackendV2 and I don't think pulse simulation for the backends is that valuable in practice it's pretty limited on what you can simulate. If you want to put something in the docstring for those missing pulse defaults that would be fine but I wouldn't think it's required.